packages/web-components/src/components/ic-toast/ic-toast.tsx

```diff
@@ -0,0 +1,305 @@
+    const found = this.interactiveElements.some((el, index) => {
+      if (!this.isActive(el)) return false;
+      currentIndex = index;
+      return true;
+    });
+
+    if (!found) return first;
+    return this.interactiveElements[currentIndex + (isBackwards ? -1 : 1)];
+  }
+
+  private isActive(targetEl: HTMLElement): boolean {
+    if (targetEl === this.el) return !!this.el.shadowRoot.activeElement;
+    return document.activeElement === targetEl;
+  }
+
+  private focusInteractiveEl(el: ActionAreaElementTypes): void {
+    if ("setFocus" in el) el.setFocus();
+    else (el.shadowRoot.firstElementChild as HTMLElement)?.focus();
+  }
+
+  componentWillLoad() {
+    this.handleLongText(
+      this.heading.length > TOAST_HEADING_CHAR_LIMIT,
+      this.message?.length > TOAST_MESSAGE_CHAR_LIMIT
+    );
+
+    if (this.autoDismissTimeout < 5000) this.autoDismissTimeout = 5000;
+
+    this.isManual = this.dismissMode === "manual";
+
+    if (this.variant === "neutral") {
+      this.neutralVariantLabel =
+        this.neutralIconAriaLabel ?? ICONS[this.variant].ariaLabel;
+    }
+  }
+
+  componentDidLoad() {
+    const actionContent = getSlot(this.el, "action") as ActionAreaElementTypes;
+    const dismissButton = this.el.shadowRoot.querySelector("ic-button");
+    if (actionContent) this.interactiveElements.push(actionContent);
+    if (dismissButton) this.interactiveElements.push(dismissButton);
+  }
+
+  render() {
+    const {
+      variant,
+      heading,
+      message,
+      visible,
+      isManual,
+      neutralVariantLabel,
+    } = this;
+    return (
+      <Host
+        class={{ ["hidden"]: !visible }}
+        role={isManual ? "dialog" : "status"}
+        aria-label={
+          variant ? neutralVariantLabel || ICONS[variant].ariaLabel : heading
+        }
+        aria-description={
+          variant
+            ? `${heading}${message !== undefined ? `. ${message}` : ""}`
+            : message
+        }
+      >
+        <div class="container">
+          {variant && (
+            <div class="toast-icon-container">
+              <div
+                class={{
+                  ["divider"]: true,
+                  [`divider-${variant}`]: true,
+                }}
+              ></div>
+              {variant === "neutral" && isSlotUsed(this.el, "neutral-icon") ? (
+                <slot name="neutral-icon" />
+              ) : (
+                <span class="toast-icon" innerHTML={ICONS[variant].icon}></span>
+              )}
+            </div>
+          )}
+          <div class="toast-content">
+            <div class="toast-message">
+              <ic-typography variant="subtitle-large">
+                <h5>{heading}</h5>
+              </ic-typography>
+              {message && (
+                <ic-typography variant="body">
+                  <p>{message}</p>
+                </ic-typography>
+              )}
+            </div>
+            {isSlotUsed(this.el, "action") && isManual && (
```

Comment:
The guidance on the site currently says "toasts that contain an action or link can’t be set to auto-dismiss."
Therefore isManual should always be set to `true` if the actions slot is being used

Reply:
^^ This relates more to line 231

The ACs also mention the point above very briefly - "For any toast with an action (therefore can't be auto-dismiss)"